main.go

```diff
@@ -129,34 +161,58 @@ func buildRelease(client *github.Client, owner string, repo string, tag string)
 	return release, nil
 }
 
-func getRelease(client *github.Client, owner string, repo string, tag string) (*github.RepositoryRelease, error) {
-	release, _, err := client.Repositories.GetReleaseByTag(owner, repo, tag)
+func (rc *releaseClient) getRelease() (*github.RepositoryRelease, error) {
+	release, _, err := rc.Client.Repositories.GetReleaseByTag(rc.Owner, rc.Repo, rc.Tag)
 	if err != nil {
-		return nil, fmt.Errorf("Release %s not found", tag)
+		return nil, fmt.Errorf("Release %s not found", rc.Tag)
 	}
 
-	fmt.Printf("Successfully retrieved %s release\n", tag)
+	fmt.Printf("Successfully retrieved %s release\n", rc.Tag)
 	return release, nil
 }
 
-func newRelease(client *github.Client, owner string, repo string, tag string) (*github.RepositoryRelease, error) {
-	rr := &github.RepositoryRelease{TagName: github.String(tag)}
-	release, _, err := client.Repositories.CreateRelease(owner, repo, rr)
+func (rc *releaseClient) newRelease() (*github.RepositoryRelease, error) {
+	rr := &github.RepositoryRelease{
+		TagName: github.String(rc.Tag),
+		Draft:   &rc.Draft,
+	}
+	release, _, err := rc.Client.Repositories.CreateRelease(rc.Owner, rc.Repo, rr)
 	if err != nil {
 		return nil, fmt.Errorf("Failed to create release: %s", err)
 	}
 
-	fmt.Printf("Successfully created %s release\n", tag)
+	fmt.Printf("Successfully created %s release\n", rc.Tag)
 	return release, nil
 }
 
-func uploadFiles(client *github.Client, owner string, repo string, id int, files []string) error {
-	assets, _, err := client.Repositories.ListReleaseAssets(owner, repo, id, &github.ListOptions{})
+func (rc *releaseClient) uploadFiles(id int, files []string) error {
+	assets, _, err := rc.Client.Repositories.ListReleaseAssets(rc.Owner, rc.Repo, id, &github.ListOptions{})
 	if err != nil {
 		return fmt.Errorf("Failed to fetch existing assets: %s", err)
 	}
 
+	var uploadFiles []string
+files:
 	for _, file := range files {
+		for _, asset := range assets {
+			if *asset.Name == path.Base(file) {
+				switch rc.FileExists {
+				case "overwrite":
+					// do nothing
+				case "fail":
+					return fmt.Errorf("Asset file %s already exists", path.Base(file))
+				case "skip":
+					fmt.Printf("Skipping pre-existing %s artifact\n", *asset.Name)
+					continue files
```

Comment:
You know,, this isnt an uncommon pattern in go...

https://github.com/golang/go/blob/08594ac7c7e123d4aa46f60690da0a7e4034f4e9/src/runtime/trace.go#L694

https://github.com/golang/go/blob/33a9a98e4d5893699749f75334c651b2adcecfb9/src/go/types/expr.go#L1056

...